### DISCUSSION TITLE
Does MultiSelectionComboBox support grouping?

### DISCUSSION THREAD

**Discussion by Filipsi** [MahApps/MahApps.Metro#4405] [Q&A]
Does MultiSelectionComboBox support item grouping and GroupStyle?
When I set ItemsSource to a CollectionViewSource, the control ends up in [custom text](https://mahapps.com/docs/controls/MultiSelectionComboBox#custom-text) state even trough IsEditable is set to false.

```xml

<UserControl.Resources>
    <CollectionViewSource
        x:Key="GroupedAvailableTargets"
        Source="{Binding AvailableTargets}">

        <CollectionViewSource.GroupDescriptions>
            <PropertyGroupDescription PropertyName="Category" />
        </CollectionViewSource.GroupDescriptions>
    </CollectionViewSource>
</UserControl.Resources>

<mah:MultiSelectionComboBox
    mah:MultiSelectorHelper.SelectedItems="{Binding SelectedTargets}"
    ItemContainerStyle="{StaticResource MahApps.Styles.MultiSelectionComboBoxItem.CheckBox}"
    ItemsSource="{Binding Source={StaticResource GroupedAvailableTargets}}"
    SelectionMode="Multiple">
```

![example](https://github.com/MahApps/MahApps.Metro/assets/5103404/9d9b0f9b-7806-40ce-9890-93af19ea5eed)

**Comment by timunie** [comment]
That is a really good question. I think it doesn't support it out of the box, but probably it can be added quite easily. For now, I'd suggest to copy the original Style and add grouping to the ListBox inside the PopUp:

https://github.com/MahApps/MahApps.Metro/blob/3fce0224bcdc62112dabecbb3757a438559bf86e/src/MahApps.Metro/Themes/MultiSelectionComboBox.xaml#L461-L476

Good luck 👍
